### PR TITLE
fix: resolve incorrect redirect to 404 page on search fields

### LINKED ISSUE #3949: Bug: Search fields redirects to  404 error page
### Describe the bug

https://github.com/user-attachments/assets/b050d140-4542-4733-b42e-74cec89fe998

### Steps to reproduce

1. Go to Explore
2. Search any repositories 
3. You will be directed to error page

### PR DESCRIPTION
**Pull request by Mandeep56Singh**
fix : resolve incorrect redirect to 404 page on search fields

Closes #3949

## Description

The search functionality in explore was redirecting users to a 404 error page
when they entered a query. This was caused by incorrect URL routing
in the search handle function

This commit fixes the routing logic by ensuring the search terms
are correctly encoded, thus preventing the 404 error. 

## Mobile & Desktop Screenshots/Recordings

https://github.com/user-attachments/assets/13ba7163-a350-4ace-8571-163aea59db28

## Tier (staff will fill in)

- [ ] Tier 1
- [ ] Tier 2
- [ ] Tier 3
- [ ] Tier 4

## [optional] What gif best describes this PR or how it makes you feel?

Really sorry due to prettier some lines sorted unintentionally

### CODE REVIEW

**Review by brandonroberts — APPROVED** [review]
Thanks! 🍕

**Review by jpmcb — APPROVED** [review]

File: components/organisms/Repositories/repositories.tsx [hunk 1]
```diff
@@ -91,7 +91,7 @@ export default function Repositories({ repositories, showSearch = true, personal
       return router.push(`/${topic}/${toolName}`);
```
Comment (brandonroberts) [inline components/organisms/Repositories/repositories.tsx:91]:
```suggestion
       return router.push(`/explore/topic/${topic}/${toolName}`);
```

**Review by brandonroberts — CHANGES_REQUESTED** [review]

**Comment by Mandeep56Singh** [comment]
@brandonroberts  sorry it  took look to apply changes, I am quite knew to open source, So I didn't knew what to do.

**Comment by brandonroberts** [comment]
@Mandeep56Singh no problem, thanks

**Review by brandonroberts — APPROVED** [review]

**Pull request merged**